tsconfig.json

```diff
@@ -0,0 +1,14 @@
+{
+  "compilerOptions": {
+    "module": "commonjs",
+    "sourceMap": true,
+    "target": "es5",
+    "outDir": "lib",
+    "declaration": true
+  },
+  "files": [
+    "src/mediator.ts",
```

Comment:
Any reason to not do "include" with "src/**/*" instead of "files"?

Also, if possible, I'd recommend setting strict null checks to true. It might cause a lot of current code to not compile, so you'll have to weigh whether it's worth it, but it's a really good option to set if possible.